tests/director/api/package_api_test.go

```diff
@@ -241,3 +241,236 @@ func TestDocumentsInPackage(t *testing.T) {
+}
+
+func TestUpdatePackage(t *testing.T) {
+	ctx := context.Background()
+
+	application := registerApplication(t, ctx, "app-test-package")
+	defer unregisterApplication(t, application.ID)
+
+	pkg := createPackage(t, ctx, application.ID, "pkg-app-1")
+	defer deletePackage(t, ctx, pkg.ID)
+
+	pkgUpdateInput := fixPackageUpdateInput("pkg-app-1-up")
+	pkgUpdate, err := tc.graphqlizer.PackageUpdateInputToGQL(pkgUpdateInput)
+	require.NoError(t, err)
+
+	updatePkgReq := fixUpdatePackageRequest(pkg.ID, pkgUpdate)
+	output := graphql.Package{}
+
+	// WHEN
+	t.Log("Update package")
+	err = tc.RunOperation(ctx, updatePkgReq, &output)
+
+	// THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, output.ID)
+
+	require.NotEmpty(t, output.Name)
+	saveExample(t, updatePkgReq.Query(), "update package")
+}
+
+func TestDeletePackage(t *testing.T) {
+	ctx := context.Background()
+
+	application := registerApplication(t, ctx, "app-test-package")
+	defer unregisterApplication(t, application.ID)
+
+	pkg := createPackage(t, ctx, application.ID, "pkg-app-1")
+
+	pkdDeleteReq := fixDeletePackageRequest(pkg.ID)
+	output := graphql.Package{}
+
+	// WHEN
+	t.Log("Delete package")
+	err := tc.RunOperation(ctx, pkdDeleteReq, &output)
+
+	// THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, output.ID)
+
+	require.NotEmpty(t, output.Name)
+	saveExample(t, pkdDeleteReq.Query(), "delete package")
+}
+
+func TestRequestPackageInstanceAuthCreation(t *testing.T) {
+	ctx := context.Background()
+
+	application := registerApplication(t, ctx, "app-test-package")
+	defer unregisterApplication(t, application.ID)
+
+	pkg := createPackage(t, ctx, application.ID, "pkg-app-1")
+	defer deletePackage(t, ctx, pkg.ID)
+
+	authCtx, inputParams := fixPackageInstanceAuthContextAndInputParams(t)
+	pkgInstanceAuthRequestInput := fixPackageInstanceAuthRequestInput(authCtx, inputParams)
+	pkgInstanceAuthRequestInputStr, err := tc.graphqlizer.PackageInstanceAuthRequestInputToGQL(pkgInstanceAuthRequestInput)
+	require.NoError(t, err)
+
+	pkgInstanceAuthCreationRequestReq := fixRequestPackageInstanceAuthCreationRequest(pkg.ID, pkgInstanceAuthRequestInputStr)
+	output := graphql.PackageInstanceAuth{}
+
+	// WHEN
+	t.Log("Request package instance auth creation")
+	err = tc.RunOperation(ctx, pkgInstanceAuthCreationRequestReq, &output)
+
+	// THEN
+	require.NoError(t, err)
+	assertPackageInstanceAuth(t, pkgInstanceAuthRequestInput, output)
+
+	saveExample(t, pkgInstanceAuthCreationRequestReq.Query(), "request package instance auth creation")
+
+	// Fetch Application with packages
+	packagesForApplicationReq := fixPackagesRequest(application.ID)
+	pkgFromAPI := graphql.ApplicationExt{}
+
+	err = tc.RunOperation(ctx, packagesForApplicationReq, &pkgFromAPI)
+	require.NoError(t, err)
+
+	// Assert the package instance auths exists
+	require.Equal(t, 1, len(pkgFromAPI.Packages.Data))
+	require.Equal(t, 1, len(pkgFromAPI.Packages.Data[0].InstanceAuths))
```

Comment:
Hmm, I think it would be great to test two main flows:
- `defaultInstanceAuth` provided:  `requestPackageInstanceAuthCreation` results in `PackageInstanceAuth` SUCCEEDED status with provided credential (copied from `defaultInstanceAuth`)
- `defaultInstanceAuth` not provided: `requestPackageInstanceAuthCreation` results in `PackageInstanceAuth` PENDING status, `setPackageInstanceAuth` changes its status to SUCCEEDED with provided credential

Reply:
So basically, as we have a separate test for `setPackageInstanceAuth`, please test here the flow with defaultInstanceAuth provided. Thanks.